### DISCUSSION TITLE
TMUX how-to?

### DISCUSSION THREAD

**Discussion by vorburger** [PatrickF1/fzf.fish#111] [Q&A] [answered]
https://github.com/jethrokuan/fzf#alternatives _lacks Tmux support_ 

https://github.com/PatrickF1/fzf.fish/wiki/Prior-Art _chose not to implement Tmux support, because users can easily add support externally themselves_

How do you get Ctrl-F etc. working inside TMUX? 😺 

https://github.com/PatrickF1/fzf.fish/wiki/Troubleshooting (or https://github.com/PatrickF1/fzf.fish/wiki/FAQ) could document it.

**Comment by vorburger** [comment]
It actually appears to work just fine out of the box under TMUX. I'm now just curious what was meant by _add support externally themselves_.

**Comment by PatrickF1** [comment, marked as answer]
Hi @vorburger! Welcome!

> what was meant by add support externally themselves.

So my understanding is that Tmux users often want to open fzf in a new pane (why? not sure). And to do that, they need to call a wrapper. It's documented here: https://github.com/junegunn/fzf#fzf-tmux-script. Seems like a popular thing. Jethrokuan/fzf implements it: https://github.com/jethrokuan/fzf/blob/8ee7cf502637a9dd7d0cd96ead67c631a25e49d9/functions/__fzfcmd.fish
This user forked my repo and changed it: https://github.com/insidexino/fzf.fish/commit/473bf33cf36bf52c7743c0c3c31b90854b3b78d5

TBH I'm not even a Tmux user and I've only used it briefly (and decided it wasn't for me). However, I suspect that we can accomplish the same thing without changing the source code. All you need to do is put in your config.fish something like this
```fish
function fzf --wraps=fzf --description="Use fzf-tmux if in tmux session"
  if set --query TMUX
    fzf-tmux $argv
  else
    command fzf $argv
end
```
See https://unix.stackexchange.com/questions/10689/how-can-i-tell-if-im-in-a-tmux-session-from-a-bash-script
Does it work? I dunno...would you like to help me test it? That way I can document it in the FAQ as you suggested.
I've added it to the Wiki for now.
